src/Test/StrictCheck/Shaped.hs

```diff
@@ -269,16 +269,16 @@ foldM alg = alg <=< traverse (translateA @a (foldM alg)) . unwrap
 unfold :: forall a f g. (Functor g, Shaped a)
        => (forall x. Shaped x => f x -> g (Shape x f))
        -> f a -> g % a
-unfold coalg = Wrap . fmap (translate @a (unfold coalg)) . coalg
+unfold coalg = Wrap . fmap (translate (unfold coalg)) . coalg
 
 -- | The 'Monad' version of 'unfold'; unfolds an interleaved structure
 -- effectfully.
 unfoldM :: forall a m f g. (Traversable g, Shaped a, Monad m)
          => (forall x. Shaped x => f x -> m (g (Shape x f)))
          -> f a -> m (g % a)
-unfoldM coalg = fmap Wrap . traverse (translateA @a (unfoldM coalg)) <=< coalg
+unfoldM coalg = fmap Wrap . traverse (translateA (unfoldM coalg)) <=< coalg
 
--- TODO: mapM, foldM, unfoldM, ...
+-- TODO: mapM, ...
```

Comment:
`mapM` is called "translateA` here. I think I should rename it...

Reply:
I would have expected `mapM` to have a type similar to `(forall x. f x -> m (g x)) -> f % a -> m (g % a)`, similar to `foldM` and `unzipWithM`. But that can already be archived rather easily with `foldM` and `unfoldM`. `foldM` requires `f` to be traversable and `unfoldM` requires `g` to be traversable, so they both versions are useful. So I think the name `translateA` is better than `mapM`. It is also not taken by `Prelude`.